src/detection.c

```diff
@@ -226,3 +218,39 @@ float get_side_right_distance(void)
 {
 	return distance[SENSOR_SIDE_RIGHT_ID];
 }
+
+/**
+ * @brief Calculate and return the side sensors error.
+ *
+ * Taking into account that the walls are parallel to the robot, this function
+ * returns the distance that the robot is moved from the center parallel line
+ * of the maze.
+ */
+float get_side_sensors_error(void)
+{
+	float sensors_error;
+
+	if ((distance[SENSOR_SIDE_LEFT_ID] > MIDDLE_MAZE_DISTANCE) &&
+	    (distance[SENSOR_SIDE_RIGHT_ID] < MIDDLE_MAZE_DISTANCE)) {
+		sensors_error =
+		    distance[SENSOR_SIDE_RIGHT_ID] - MIDDLE_MAZE_DISTANCE;
+	} else if ((distance[SENSOR_SIDE_RIGHT_ID] > MIDDLE_MAZE_DISTANCE) &&
+		   (distance[SENSOR_SIDE_LEFT_ID] < MIDDLE_MAZE_DISTANCE)) {
+		sensors_error =
+		    MIDDLE_MAZE_DISTANCE - distance[SENSOR_SIDE_LEFT_ID];
+	} else {
+		sensors_error = 0;
+	}
+	return sensors_error;
+}
```

Comment:
I'd use the `if (...) return;` syntax here, but I guess that's a matter of preference... :stuck_out_tongue_winking_eye: 

See for example:

```c
float get_side_sensors_error(void)
{
	if ((distance[SENSOR_SIDE_LEFT_ID] > MIDDLE_MAZE_DISTANCE) &&
	    (distance[SENSOR_SIDE_RIGHT_ID] < MIDDLE_MAZE_DISTANCE))
		return distance[SENSOR_SIDE_RIGHT_ID] - MIDDLE_MAZE_DISTANCE;
	if ((distance[SENSOR_SIDE_RIGHT_ID] > MIDDLE_MAZE_DISTANCE) &&
		   (distance[SENSOR_SIDE_LEFT_ID] < MIDDLE_MAZE_DISTANCE))
		return MIDDLE_MAZE_DISTANCE - distance[SENSOR_SIDE_LEFT_ID];
	return 0;
}
```

Reply:
I prefer the else if, else. 😇